plugin/evm/limit_order.go

```diff
@@ -91,14 +115,55 @@ func (lop *limitOrderProcesser) RunMatchingEngine() {
 	}
 }
 
+func (lop *limitOrderProcesser) runLiquidations(market limitorders.Market, longOrders []limitorders.LimitOrder, shortOrders []limitorders.LimitOrder) (filteredLongOrder []limitorders.LimitOrder, filteredShortOrder []limitorders.LimitOrder) {
+	oraclePrice := big.NewInt(20 * 10e6) // @todo: get it from the oracle
+
+	liquidablePositions := lop.memoryDb.GetLiquidableTraders(market, oraclePrice)
+
+	for i, liquidable := range liquidablePositions {
+		var oppositeOrders []limitorders.LimitOrder
+		switch liquidable.PositionType {
+		case "long":
+			oppositeOrders = shortOrders
+		case "short":
+			oppositeOrders = longOrders
+		}
+		if len(oppositeOrders) == 0 {
+			log.Error("no matching order found for liquidation", "trader", liquidable.Address.String(), "size", liquidable.Size)
+			continue // so that all other liquidable positions get logged
+		}
+		for j, oppositeOrder := range oppositeOrders {
+			if liquidable.GetUnfilledSize().Sign() == 0 {
+				break
+			}
+			fillAmount := utils.BigIntMinAbs(liquidable.GetUnfilledSize(), oppositeOrder.GetUnFilledBaseAssetQuantity())
+			if fillAmount.Sign() == 0 {
+				continue
+			}
+			lop.limitOrderTxProcessor.ExecuteLiquidation(liquidable.Address, oppositeOrder, fillAmount)
+
+			switch liquidable.PositionType {
+			case "long":
+				oppositeOrders[j].FilledBaseAssetQuantity.Sub(oppositeOrders[j].FilledBaseAssetQuantity, fillAmount)
+				liquidablePositions[i].FilledSize.Add(liquidablePositions[i].FilledSize, fillAmount)
+			case "short":
+				oppositeOrders[j].FilledBaseAssetQuantity.Add(oppositeOrders[j].FilledBaseAssetQuantity, fillAmount)
+				liquidablePositions[i].FilledSize.Sub(liquidablePositions[i].FilledSize, fillAmount)
```

Comment:
I think this is incorrect because `liq.Size` is an absolute value and
```
func (liq LiquidablePosition) GetUnfilledSize() *big.Int {
	return big.NewInt(0).Sub(liq.Size, liq.FilledSize)
}
```
How about keeping this but making liq.Size signed?